### ISSUE TITLE
Problem with Steam Codes

### ISSUE THREAD

**Issue by dommilosz** [Authenticator-Extension/Authenticator#405]
I have problem with adding steam account.
When i try to input shared_secret it tells me that secret is incorrect
![Bez tytułu](https://user-images.githubusercontent.com/54410730/69737539-d653a900-1134-11ea-85a4-7a6b88a95890.png)

**Platform:** Win 10
 - Browser: Firefox
 - Browser Version: 70.0.1

**Comment by mymindstorm** [comment]
Are you putting in `"shared_secret":"#############################",`? You should only enter the text where the hashtags are.

**Comment by dommilosz** [comment]
yes

**Comment by dommilosz** [comment]
only text in quotation marks

**Comment by mymindstorm** [comment]
Could you run the below code in the console and paste the results here?

<details>
<summary>Snippet</summary>


```js
let pattern1 = /([^a-z2-7])+/i;
let pattern2 = /[^0-9a-f]+/i;
let secret = prompt("Secret");
secret = secret.replace(/ /g, '');

let result1 = pattern1.exec(secret);
let result2 = pattern2.exec(secret)

if (result1 && result2) {
    console.log("b32 invalid:", result1.toString());
	console.log("hex invalid:", result2.toString());
}
```

</details>

**Comment by dommilosz** [comment]
undefined
b32 invalid: +8,8 debugger eval code:10:13
hex invalid: MpKLJ debugger eval code:11:10

**Comment by mymindstorm** [comment]
@dommilosz I think your secret is in base64, see if you can convert it to base32 or hex. I don't have access to a shared_secret, so I can't investigate this that easily.

**Comment by AlexApps99** [comment]
@mymindstorm @dommilosz 
See #382 for why it isn't working
tl;dr you need to copy
"uri":"otpauth://totp/Steam:your_steam_username?secret=**your_secret_key**&issuer=Steam"
instead of
"shared_secret":"**your_secret_key**"

**Comment by 1024mb** [comment]
I have a problem with Steam codes too. I've imported from Aegis and the Steam import is giving me number codes when it should have letters thus Steam not accepting those codes.

This is the Steam line:

`otpauth://steam/Steam%3AUSERNAME?period=30&digits=5&algorithm=SHA1&secret=SECRET&issuer=Steam`
